pyomo/core/base/constraint.py

```diff
@@ -772,6 +770,10 @@ def construct(self, data=None):
         if is_debug_set(logger):
             logger.debug("Constructing constraint %s" % (self.name))
 
+        if self._anonymous_sets is not None:
```

Comment:
In this case, no.  I partially misspoke in my reply to Miranda (I missed which part of the documentation we were talking about).  `_anonymous_sets` will exist on all `IndexedComponent` classes - I ended up taking the conditional definition of the attribute out (although it can be None [and not a `ComponentSet`]).